src/Forms/TreeDropdownField.php

```diff
@@ -519,13 +520,20 @@ public function tree(HTTPRequest $request)
 
         // Allow to pass values to be selected within the ajax request
         $value = $request->requestVar('forceValue') ?: $this->value;
-        if ($value && ($values = preg_split('/,\s*/', $value ?? ''))) {
+        if ($value instanceof SS_List) {
+            $values = $value;
```

Comment:
There were cases where an `SS_List` was being passed as the value - so just iterate over that directly.